### PR TITLE
- addd socketcam support issue #52

### LINKED ISSUE #52: Add SocketCam support

### PR DESCRIPTION
**Pull request by ShawnDhanani**

### CODE REVIEW

File: app/src/main/java/com/socketmobile/stockcount/ui/EditActivity.kt [hunk 1]
```diff
@@ -223,8 +263,16 @@ class EditActivity : AppCompatActivity() {
     private fun triggerDevices() {
         val readyDeviceGuids = deviceStateMap.filter { entry -> entry.value.intValue() == DeviceState.READY }.keys
         val readyDevices = deviceClientMap.filter { entry -> readyDeviceGuids.contains(entry.key) }.values
```
Comment (doubleo2) [inline app/src/main/java/com/socketmobile/stockcount/ui/EditActivity.kt:265]:
```suggestion
        val readyDevices = deviceStateMap.takeIf { intValue() == DeviceState.READY }.mapNotNull { deviceClientMap[it.key] }
```

Not your code, but can be cleaned up

File: app/src/main/java/com/socketmobile/stockcount/ui/EditActivity.kt [hunk 2]
```diff
@@ -254,10 +302,17 @@ class EditActivity : AppCompatActivity() {
...
-        deviceClientMap[deviceGuid] = event.device
+        val device = event.device
+        var state = event.state
+        val scannerStatus = state.intValue()
+        val deviceGuid = device.deviceGuid
+        deviceStateMap[deviceGuid] = state
+        deviceClientMap[deviceGuid] = device
+
+        if (!isSocketCamDevice(device.deviceName)) {
+            stopSocketCamExtension()
```
Comment (doubleo2) [inline app/src/main/java/com/socketmobile/stockcount/ui/EditActivity.kt:313]: When you stop the extension, do you receive a device removal event for the SocketCam reader?

File: app/src/main/java/com/socketmobile/stockcount/helper/CaptureHelper.kt [hunk 3]
```diff
@@ -1,4 +1,6 @@
 /**  Copyright © 2018 Socket Mobile, Inc. */
 
 package com.socketmobile.stockcount.helper
-
+fun isSocketCamDevice(deviceName: String): Boolean {
+    return deviceName == "SocketCam C820" || deviceName == "SocketCam C860"
+}
```
Comment (doubleo2) [inline app/src/main/java/com/socketmobile/stockcount/helper/CaptureHelper.kt:6]: How about making this a DeviceClient extension? In any case it shouldn't be based on the device name

```suggestion
fun DeviceClient.isSocketCamDevice(): Boolean {
    return deviceType == kModelSocketCamC820
}
```

File: app/src/main/java/com/socketmobile/stockcount/ui/EditActivity.kt (hunk 2 above)
Comment (ShawnDhanani) [reply app/src/main/java/com/socketmobile/stockcount/ui/EditActivity.kt:315]: yes, receive device removal event after stop the extension

**Pull request merged**